src/permutation-switch.js

```diff
@@ -0,0 +1,82 @@
+(function () {
+  'use strict';
+
+  /**
+   * `permutation-switch`
+   *
+   * This provides an expression to be evaluated by child predicate permutations
+   */
+  function PermutationSwitchDirective ($parse) {
+    return {
+      restrict: 'AE',
+      require: 'permutationSwitch',
+      priority: 500,
+      controller: ['$scope', function NgPermutationSwitchCtrl () {
+        this.permutations = [];
+
+        this.addPermutation = function (permutation) {
+          this.permutations.push(permutation);
+        };
+      }],
+      link: function (scope, element, attrs, permutationSwitchController) {
+        var watchExpression = attrs.permutationSwitch || attrs.on,
+            selectedPermutations = [];
+
+        // re-evaluate which transclusions to display when the main expression changes
+        scope.$watch(watchExpression, function permutationSwitchWatchAction (value) {
+          var selectedPermutation;
+
+          // destroy existing transcluded scopes and elements
+          while (selectedPermutations.length > 0) {
+            selectedPermutation = selectedPermutations.pop();
+            selectedPermutation.scope.$destroy();
+            selectedPermutation.element.remove();
+          }
+
+          // examine each child predicate and transclude the ones that come out truthy
+          angular.forEach(permutationSwitchController.permutations, function (selectedTransclude) {
+            var predicate = $parse(selectedTransclude.expression)(scope);
```

Comment:
Scope is only isolated on the directive if you define scope properties, otherwise it shares it